### PR TITLE
v2 changes of platform settings

### PR DESCRIPTION
**Pull request by muhammad-ali-e**
## What
- v2 changes of platform settings
  - Updated the application name.
  - Imported packages from v2 applications instead of v1.
  - Retrieved organization information from UserContext or Session instead of connection.tenant.

## Why

- The changes are necessary to align with the updated architecture and to ensure compatibility with the v2 applications.
- To streamline the handling of organization contexts without relying on django_tenants.

## How

- Modified `platform_auth_helper.py` to update the import statements from v2.
- Adjusted `platform_auth_service.py` to import from v2 applications and updated the methods of get_organization by using user context.
- Updated `serializers.py` to reflect import changes.
- Changed the application name in `apps.py`.
- Updated `views.py` for consistency with the new application structure.

### Changed files:

- platform_auth_helper.py (package import changes only)
- platform_auth_service.py
- serializers.py (import changes only)
- apps.py (app name change only)
- views.py

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why. (PS: Admins do not merge the PR without this section filled)

- The new methods are self-contained ensuring no impact on current functionality.
- Also under feature flag `multi_tenancy_v2`.

## Database Migrations

- 

## Env Config

- 

## Relevant Docs

-

## Related Issues or PRs

1. https://github.com/Zipstack/unstract/pull/422
2. https://github.com/Zipstack/unstract/pull/430
3. https://github.com/Zipstack/unstract/pull/431
4. https://github.com/Zipstack/unstract/pull/432

## Dependencies Versions

-

## Notes on Testing

-

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: backend/platform_settings_v2/platform_auth_helper.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        auth_controller = AuthenticationController()
+        try:
+            member: OrganizationMember = (
+                auth_controller.get_organization_members_by_user(user=user)
+            )
+        except Exception as error:
+            logger.error(
+                f"Error occurred while fetching organization for user : {error}"
+            )
+            raise error
```
Comment (ritwik-g) [inline backend/platform_settings_v2/platform_auth_helper.py:33]: @muhammad-ali-e do we need the stack and trace here? Or will raising the error again will take care of it?

File: backend/platform_settings_v2/platform_auth_helper.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    @staticmethod
+    def validate_token_count(organization: Organization) -> None:
+        if (
+            PlatformKey.objects.filter(organization=organization).count()
+            >= PLATFORM_KEY_COUNT
+        ):
+            logger.error(
+                f"Only {PLATFORM_KEY_COUNT} keys are support at a time. Count exceeded."
+            )
```
Comment (ritwik-g) [inline backend/platform_settings_v2/platform_auth_helper.py:48]: @muhammad-ali-e should we include `organization` detail in the log for more detail?

File: backend/platform_settings_v2/platform_auth_service.py [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+            platform_key.save()
+            result: dict[str, Any] = {}
+            result[PlatformServiceConstants.ID] = platform_key.id
+            result[PlatformServiceConstants.KEY_NAME] = platform_key.key_name
+            result[PlatformServiceConstants.KEY] = platform_key.key
+
+            logger.info(f"platform_key is generated for {organization.id}")
+            return result
+        except IntegrityError as error:
+            logger.error(f"Integrity error - failed to generate platform key : {error}")
```
Comment (ritwik-g) [inline backend/platform_settings_v2/platform_auth_service.py:77]: @muhammad-ali-e should we include org detail here for better context?

File: backend/platform_settings_v2/platform_auth_service.py [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+            raise DuplicateData(
+                f"{ErrorMessage.KEY_EXIST}, \
+                            {ErrorMessage.DUPLICATE_API}"
+            )
+        except Exception as error:
+            logger.error(f"Failed to generate platform key : {error}")
+            raise InternalServiceError()
+
+    @staticmethod
+    def delete_platform_key(id: str) -> None:
```
Comment (ritwik-g) [inline backend/platform_settings_v2/platform_auth_service.py:87]: @muhammad-ali-e should we include other details like org or something as well to make sure the key deletions happens only for the respected org? So we could even modify the query to include org? Will that make sense here?

File: backend/platform_settings_v2/platform_auth_service.py [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+        """This is a delete operation. Use this function only if you know what
+        you are doing.
+
+        Args:
+            id (str): _description_
+
+        Raises:
+            error: _description_
+        """
```
Comment (ritwik-g) [inline backend/platform_settings_v2/platform_auth_service.py:96]: @muhammad-ali-e please improve the documentation here. `This is a delete operation. Use this function only if you know what you are doing.`  - This message doesn't feel appropriate. 

Also add missing `_description_`

File: backend/platform_settings_v2/platform_auth_service.py [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+            return result
+        except IntegrityError as error:
+            logger.error(f"Integrity error - failed to refresh platform key : {error}")
+            raise DuplicateData(
+                f"{ErrorMessage.KEY_EXIST}, \
+                            {ErrorMessage.DUPLICATE_API}"
+            )
+        except Exception as error:
+            logger.error(f"Failed to refresh platform key : {error}")
+            raise InternalServiceError()
```
Comment (ritwik-g) [inline backend/platform_settings_v2/platform_auth_service.py:142]: May be add more contexts in to the error messages to figure out which org or platform_key id etc,.

**Review by ritwik-g — APPROVED** [review]
Overall looks okay. Would be better if we enrich error logs.

File: backend/platform_settings_v2/platform_auth_helper.py (hunk 1 above)
Comment (muhammad-ali-e) [reply backend/platform_settings_v2/platform_auth_helper.py]: Removed unwanted handling generic exception

File: backend/platform_settings_v2/platform_auth_helper.py (hunk 2 above)
Comment (muhammad-ali-e) [reply backend/platform_settings_v2/platform_auth_helper.py]: Updated

File: backend/platform_settings_v2/platform_auth_service.py (hunk 3 above)
Comment (muhammad-ali-e) [reply backend/platform_settings_v2/platform_auth_service.py]: Updated

File: backend/platform_settings_v2/platform_auth_service.py (hunk 5 above)
Comment (muhammad-ali-e) [reply backend/platform_settings_v2/platform_auth_service.py]: Updated

File: backend/platform_settings_v2/platform_auth_service.py (hunk 4 above)
Comment (muhammad-ali-e) [reply backend/platform_settings_v2/platform_auth_service.py:87]: We can add it in logs without accepting the params. Can be done by later. Anyway the Id here is primary key, So I think it not needed here for an extra validation.

File: backend/platform_settings_v2/platform_auth_service.py (hunk 6 above)
Comment (muhammad-ali-e) [reply backend/platform_settings_v2/platform_auth_service.py]: Removed unwanted generic excpetion handler

**Comment by muhammad-ali-e** [comment]
> Overall looks okay. Would be better if we enrich error logs.

Addressed All comments

File: backend/platform_settings_v2/exceptions.py [hunk 7]
```diff
@@ -0,0 +1,49 @@
+from typing import Optional
+
+from rest_framework.exceptions import APIException
+
+
+class InternalServiceError(APIException):
+    status_code = 500
+    default_detail = "Internal error occurred while platform key operations."
```
Comment (chandrasekharan-zipstack) [inline backend/platform_settings_v2/exceptions.py:8]:
```suggestion
    default_detail = "Internal error occurred while performing platform key operations."
```

File: backend/platform_settings_v2/exceptions.py [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+
+class UserForbidden(APIException):
+    status_code = 403
+    default_detail = (
+        "User is forbidden from performing this action. Please contact admin"
```
Comment (chandrasekharan-zipstack) [inline backend/platform_settings_v2/exceptions.py:14]:
```suggestion
        "User is forbidden from performing this action. Please contact admin."
```
NIT: Let's add periods to exception messages shown to the user

**Review by vishnuszipstack — APPROVED** [review]
Left a NIT comment, others looks good.

File: backend/platform_settings_v2/platform_auth_service.py [hunk 9]
```diff
@@ -0,0 +1,235 @@
...
+            platform_key.modified_by = user
+            if action == PlatformServiceConstants.ACTIVATE:
+                active_keys: list[PlatformKey] = PlatformKey.objects.filter(
+                    is_active=True, organization=organization
+                ).all()
+                # Deactivates all keys
+                for key in active_keys:
+                    key.is_active = False
+                    key.modified_by = user
+                    key.save()
```
[the remark this answers is not in the document]
Comment (muhammad-ali-e) [reply backend/platform_settings_v2/platform_auth_service.py]: updated  this function

File: backend/platform_settings_v2/platform_auth_service.py (hunk 4 above)
Comment (ritwik-g) [reply backend/platform_settings_v2/platform_auth_service.py:88]: The only reason to suggest the org also to be passed is to make sure we delete the PK of indented org alone even if we are deleting using the primarykey/

May be we can add a TODO for revisit this later

Comment (muhammad-ali-e) [reply backend/platform_settings_v2/platform_auth_service.py:88]: Added a TODO comment here

**Comment by hari-kuriakose** [comment]
We can take care of duplication when we delete v1 eventually. Bypassing.

**Pull request merged**